### PR TITLE
Replaced Jackson by combination of GSON and AutoValue extension. 

### PR DESCRIPTION
**Pull request by ArazAbishov**
Reducing method count by replacing Jackson with GSON. Please, see: https://github.com/artem-zinnatullin/qualitymatters/issues/136

### CODE REVIEW

**Comment by lenguyenthanh** [comment]
@ArazAbishov could you explain why we should switch from Gson to Jackson?

**Comment by ArazAbishov** [comment]
It is quite strange that tests are failing on CI servers, while locally all of them are passing.

**Comment by artem-zinnatullin** [comment]
@danielgomezrico huh, this PR failed with OOM on both CIs

https://travis-ci.org/artem-zinnatullin/qualitymatters/builds/166865076
https://circleci.com/gh/artem-zinnatullin/qualitymatters/189?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: app/src/integrationTests/java/com/artemzin/qualitymatters/integration_tests/api/other/GsonTypeAdapterFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.junit.runner.RunWith;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(QualityMattersIntegrationRobolectricTestRunner.class)
+public class GsonTypeAdapterFactoryTest {
+
+    // check if factory is aware of Item type and returns a TypeAdapter
+    @Test
+    public void supportsItemTypeAdapter() {
```
Comment (artem-zinnatullin) [inline app/src/integrationTests/java/com/artemzin/qualitymatters/integration_tests/api/other/GsonTypeAdapterFactoryTest.java]: hm, don't you think that `fromJson` is enough test for that?

Comment (vanniktech) [inline app/src/integrationTests/java/com/artemzin/qualitymatters/integration_tests/api/other/GsonTypeAdapterFactoryTest.java]: yeah I'd say so too

File: app/src/main/java/com/artemzin/qualitymatters/other/GsonTypeAdapterFactory.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.artemzin.qualitymatters.other;
+
+import com.artemzin.qualitymatters.api.entities.Item;
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+
+public class GsonTypeAdapterFactory implements TypeAdapterFactory {
```
Comment (ArazAbishov) [inline app/src/main/java/com/artemzin/qualitymatters/other/GsonTypeAdapterFactory.java]: You are right. I will change to the factory generated by autovalue-gson. For some reason I have overlooked it in documentation.

File: app/src/integrationTests/java/com/artemzin/qualitymatters/integration_tests/api/other/GsonTypeAdapterFactoryTest.java (hunk 1 above)
Comment (ArazAbishov) [inline app/src/integrationTests/java/com/artemzin/qualitymatters/integration_tests/api/other/GsonTypeAdapterFactoryTest.java]: Agreed. I will remove this test in next version of PR.

**Comment by ArazAbishov** [comment]
@artem-zinnatullin Should I resolve conflicts in `build.gradle`? Conflicts were introduced in the PR which was merged in lately.

**Comment by vanniktech** [comment]
Yes you'll need to resolve them. Sorry for that :D

**Comment by artem-zinnatullin** [comment]
That would be good, or I'll resolve and merge manually on my side later

**Comment by ArazAbishov** [comment]
@artem-zinnatullin Resolved conflicts.

**Comment by artem-zinnatullin** [comment]
Thanks, @ArazAbishov!

**Pull request merged**